modules/javafx.graphics/src/main/java/com/sun/prism/Image.java

```diff
@@ -200,60 +205,104 @@ public static Image fromFloatMapData(FloatBuffer pixels, int width, int height)
-     * If it necessary, it does in-place format conversion like RGBA->BGRA
+     * If possible, it does not duplicate the memory.
+     * If necessary, it does in-place format conversion like RGBA->BGRA
      *
      * @param frame ImageFrame to convert.
      * @return New Image instance.
      */
     public static Image convertImageFrame(ImageFrame frame) {
-        ByteBuffer buffer = (ByteBuffer) frame.getImageData();
         ImageStorage.ImageType type = frame.getImageType();
         int w = frame.getWidth(), h = frame.getHeight();
-        int scanBytes = frame.getStride();
+        int stride = frame.getStride();
         float ps = frame.getPixelScale();
 
-        switch (type) {
-            case GRAY:
-                return Image.fromByteGrayData(buffer, w, h, scanBytes, ps);
-
-            case RGB:
-                return Image.fromByteRgbData(buffer, w, h, scanBytes, ps);
-
-            case RGBA:
-                // Bgra => BgrePre is same operation as Rgba => RgbaPre
-                // TODO: 3D - need a way to handle pre versus non-Pre
-                ByteBgra.ToByteBgraPreConverter().convert(buffer, 0, scanBytes,
-                                                          buffer, 0, scanBytes,
-                                                          w, h);
-                /* NOBREAK */
-            case RGBA_PRE:
-                ByteRgba.ToByteBgraConverter().convert(buffer, 0, scanBytes,
-                                                       buffer, 0, scanBytes,
-                                                       w, h);
-                return Image.fromByteBgraPreData(buffer, w, h, scanBytes, ps);
-
-            case GRAY_ALPHA:
-                // TODO: 3D - need a way to handle pre versus non-Pre
-                ByteGrayAlpha.ToByteGrayAlphaPreConverter().convert(buffer, 0, scanBytes,
-                                                                    buffer, 0, scanBytes,
-                                                                    w, h);
-                /* NOBREAK */
-            case GRAY_ALPHA_PRE:
-                if (scanBytes != w * 2) {
-                    throw new AssertionError("Bad stride for GRAY_ALPHA");
-                }
-                byte newbuf[] = new byte[w * h * 4];
-                ByteGrayAlphaPre.ToByteBgraPreConverter().convert(buffer, 0, scanBytes,
-                                                                  newbuf, 0, w*4,
-                                                                  w, h);
-                return Image.fromByteBgraPreData(newbuf, w, h, ps);
-            default:
-                throw new RuntimeException("Unknown image type: " + type);
-        }
+        // GRAY, RGB, BGRA_PRE, and INT_ARGB_PRE are directly supported by Prism.
+        // We'll need to convert all other formats that we might encounter to one of the supported formats.
+        // TODO: 3D - need a way to handle pre versus non-Pre
```

Comment:
Is this TODO still relevant?

Reply:
Probably not. In any case, I don't know what the todo specifically wants us to do.